### PR TITLE
prettier 1.9 and config handling

### LINKED ISSUE #81: Respect editorconfig specs
As stated in https://github.com/prettier/prettier/issues/42 and https://github.com/prettier/prettier-atom/issues/90, it would be great to have prettier surrender to the specs defined in a project-wise `.editorconfig` file.

### LINKED ISSUE #199: Enable only if prettier is installed
Would it be possible to add an option which disables this extension if prettier is not installed in the root package json?

### LINKED ISSUE #248: Do not use editor configuration if project uses prettier
Prettier VSCode shouldn't merge the custom editor configuration with the configuration from the projects as this leads to a different style than the one from the CLI. It would be great if the custom configuration would be only used when the project doesn't use Prettier directly.

### LINKED ISSUE #273: enable stylelintIntegration with scss not work with 'prettier-stylelint'?
i am confused for  a long time, because i can't make my editor to format my scss file crollectly even by this extension.

i try myself change my vscode config for the extension so many time but not work;

i want to format my stylesheet file by my stylelint config.

but i found that when i try to launch the extension and debug that:
https://github.com/prettier/prettier-vscode/blob/8f31fa2c2af98087d9df50f04b15100f0895a303/src/PrettierEditProvider.ts#L111

why condition include `parser === 'postcss'`,   `parser` was equaled 'scss' when i launched and debugged with scss file?   

how can i change the prettier config to make it work?

### PR DESCRIPTION
**Pull request by CiGit**
* proseWrap config change
* Don't merge vscode's option in prettierConfig -- fix #248 
* dependencies bump
* Fix #273 add style parsers
+ arrowParens config
+ requireConfig config. Format only files with a prettierconfig -- fix #199
+ using resolveConfig + editorconfig -- fix #81 

- [x] Readme
- [x] Changelog
- [ ] ~~json validation for `prettierrc.json`~~ other PR

## Break

Don't handle deprecated options. Workaround: prettierconfig

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -157,18 +174,18 @@
...
+    "cross-env": "^5.1.1",
     "mocha": "^3.5.3",
-    "typescript": "^2.4.2",
-    "vscode": "^1.1.6"
+    "typescript": "^2.6.2",
+    "vscode": "^1.1.10"
   },
   "dependencies": {
+    "ignore": "^3.3.7",
+    "prettier": "1.9.0",
```
Comment (azz) [inline package.json]: 1.9.1 is out now

**Comment by RobinMalfait** [comment]
Could you also bump the version of prettier-eslint if there is a new version?

File: package.json [hunk 2]
```diff
@@ -157,18 +174,18 @@
...
+    "prettier-eslint": "^8.2.4",
```
Comment (CiGit) [inline package.json]: @RobinMalfait  8.2.0 -> 8.2.4
There is also a 8.2.5 now

Comment (RobinMalfait) [reply package.json]: Whoops, didn't see that!

**Comment by CiGit** [comment]
Deprecated old options, notably "postcss" and updated stylelint check with new styles

**Comment by CiGit** [comment]
Good for a review !

**Comment by CiGit** [comment]
Does someone see something more we would need for a 1.0.0 release?

**Pull request merged**